vizro-core/src/vizro/models/_page.py

```diff
@@ -116,7 +131,8 @@ def pre_build(self):
                     ),
                     actions=[
                         Action(
-                            id=f"{ON_PAGE_LOAD_ACTION_PREFIX}_action_{self.id}", function=_on_page_load(page_id=self.id)
+                            id=f"{ON_PAGE_LOAD_ACTION_PREFIX}_action_{self.id}",
+                            function=_on_page_load(targets=targets),
```

Comment:
Targets are calculated in advance for the `_on_page_load` action (aligned with eagerly calculating targets for the `_filter` action).